src/Pages/Media/style.ts

```diff
@@ -193,44 +451,139 @@ export const CarouselWrapper = styled.div`
     gap: 20px;
     margin: 10px;
   }
 
-  @media (min-width: 768px) and (max-width: 1023px) {
+  @media (${devices.mobileM}) {
+    grid-template-columns: repeat(1, minmax(0, 1fr));
+    gap: 20px;
+    margin: 10px;
+  }
+
+  @media (${devices.mobileL}) {
+    grid-template-columns: repeat(1, minmax(0, 1fr));
+    gap: 20px;
+    margin: 10px;
+  }
+
+  @media (${devices.tablet}) {
     grid-template-columns: repeat(1, minmax(0, 1fr));
     gap: 30px;
     margin: 10px;
   }
 
-  @media (min-width: 1024px) {
+  @media (${devices.laptop}) {
+    grid-template-columns: repeat(2, minmax(0, 1fr));
+    gap: 40px;
+    margin: 20px;
+  }
+
+  @media (${devices.laptopL}) {
+    grid-template-columns: repeat(2, minmax(0, 1fr));
+    gap: 40px;
+    margin: 20px;
+  }
+
+  @media (${devices.desktop}) {
     grid-template-columns: repeat(2, minmax(0, 1fr));
     gap: 40px;
     margin: 20px;
   }
 `;
+
 export const SpotifyWrapper = styled.iframe.attrs({
   width: 800,
   height: 500,
 })`
-  @media (max-width: 768px) {
+  @media (${devices.mobileES}) {
     width: 80%;
     height: 400px;
   }
 
-  @media (min-width: 1024px) {
-    width: 800px;
-    height: 500px;
+  @media (${devices.mobileS}) {
+    width: 80%;
+    height: 400px;
+  }
+
+  @media (${devices.mobileM}) {
+    width: 80%;
+    height: 400px;
+  }
+
+  @media (${devices.mobileL}) {
+    width: 80%;
+    height: 400px;
   }
 
-  @media (max-width: 480px) {
+  @media (${devices.tablet}) {
     width: 80%;
     height: 400px;
   }
+
+  @media (${devices.laptop}) {
+    width: 800px;
+    height: 500px;
+  }
+
+  @media (${devices.laptopL}) {
+    width: 800px;
+    height: 500px;
+  }
+
+  @media (${devices.desktop}) {
+    width: 800px;
+    height: 500px;
+  }
 `;
 
 export const LinkStyle = styled.a`
   text-decoration: none;
 `;
+
+export const opts = {
+  height: "auto",
+  width: "auto",
+
+  playerVars: {
+    autoplay: 0,
+  },
+};
+
+// Check screen size and set width and height accordingly
+if (window.matchMedia("(min-width: 2560px)").matches) {
+  opts.width = "1000px";
+  opts.height = "550px";
+} else if (window.matchMedia("(min-width: 1440px)").matches) {
+  opts.width = "600px";
+  opts.height = "400px";
+} else if (window.matchMedia("(min-width: 1024px)").matches) {
+  opts.width = "500px";
+  opts.height = "300px";
+} else if (window.matchMedia("(min-width: 768px)").matches) {
+  opts.width = "350px";
+  opts.height = "225px";
+} else if (window.matchMedia("(min-width: 425px)").matches) {
+  opts.width = "400px";
+  opts.height = "400px";
+} else if (window.matchMedia("(min-width: 375px)").matches) {
+  opts.width = "350px";
+  opts.height = "400px";
+} else if (window.matchMedia("(min-width: 320px)").matches) {
+  opts.width = "300px";
+  opts.height = "300px";
+} else if (window.matchMedia("(min-width: 100px)").matches) {
+  opts.width = "250px";
+  opts.height = "150px";
+} else {
+  opts.width = "500px";
+  opts.height = "300px";
+}
```

Comment:
```suggestion
const getOptsForYoutube = {
  // Check screen size and set width and height accordingly
  if (window.matchMedia("(min-width: 2560px)").matches) {
    return { width: "500px", height: "300px" }
  } else if (window.matchMedia("(min-width: 1440px)").matches) {
    return { width: "500px", height: "300px" }
  } else if (window.matchMedia("(min-width: 1024px)").matches) {
    return { width: "500px", height: "300px" }
  } else if (window.matchMedia("(min-width: 768px)").matches) {
    return { width: "500px", height: "300px" }
  } else if (window.matchMedia("(min-width: 425px)").matches) {
    return { width: "500px", height: "300px" }
  } else if (window.matchMedia("(min-width: 375px)").matches) {
    return { width: "500px", height: "300px" }
  } else if (window.matchMedia("(min-width: 320px)").matches) {
    return { width: "500px", height: "300px" }
  } else if (window.matchMedia("(min-width: 100px)").matches) {
    return { width: "500px", height: "300px" }
  } else {
    return { width: "500px", height: "300px" }
  }
}
```

and call function `getOptsForYoutube` from the index.ts file and then pass this value to the Youtube component.